### PR TITLE
Force keeper snapshot equality on different replicas

### PR DESCRIPTION
**Pull request by alesapin**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Review by tavplubix — APPROVED** [review]

File: src/Common/ZooKeeper/IKeeper.h [hunk 1]
```diff
@@ -34,6 +34,12 @@ struct ACL
     int32_t permissions;
     String scheme;
     String id;
+
+    bool operator<(const ACL & other) const
+    {
+        return std::tuple(permissions, scheme, id)
+            < std::tuple(other.permissions, other.scheme, other.id);
+    }
```
Comment (antonio2368) [inline src/Common/ZooKeeper/IKeeper.h:42]:
```suggestion
    auto operator<=>(const ACL &) const = default;
```
should work here
https://en.cppreference.com/w/cpp/language/default_comparisons

**Review by antonio2368 — APPROVED** [review]

**Comment by alesapin** [comment]
`explicitly defaulted three-way comparison operator is implicitly deleted` :)

**Comment by alesapin** [comment]
Cannot understand what was wrong, reverted `<=>`.

**Comment by alesapin** [comment]
Gcc failure unrelated, but let's try one more time.

**Comment by alesapin** [comment]
@merfiyio update

**Comment by alesapin** [comment]
@mergifyio update

**Comment by alesapin** [comment]
Hmm, looks like gcc failure related...

**Comment by alesapin** [comment]
I'll just temporarily turn off gcc build. Need to update it to gcc-12.

**Comment by alesapin** [comment]
`01281_group_by_limit_memory_tracking` -- flaky test.
Stress debug -- https://github.com/ClickHouse/ClickHouse/issues/37094
Stress thread/undefined -- thread fuzzer disabled in master.

**Pull request merged**